src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -46,13 +45,12 @@ public byte[] mask(byte[] input) {
 
             KeyMaskingConfig keyMaskingConfig = maskingConfig.isInAllowMode() ? maskingConfig.getDefaultConfig() : null;
             if (maskingState.jsonPathEnabled()) {
-                // Check for "$" JSONPath key.
-                keyMaskingConfig = keyMatcher.getMaskConfigIfMatched(
-                        maskingState.getMessage(),
-                        -1,
-                        -1,
-                        Collections.emptyIterator()
-                );
+                // A special case of "$" JSONPath key.
+                maskingState.expandCurrentJsonPath(keyMatcher.getJsonPathRootNode());
+                if (maskingState.getCurrentJsonPathNode().endOfWord && maskingConfig.isInAllowMode()) {
+                    return input;
+                }
+                keyMaskingConfig = keyMatcher.getMaskConfigIfMatched(maskingState.getMessage(), -1, -1, maskingState.getCurrentJsonPathNode());
             }
```

Comment:
With this amount of code handling this particular edge case (`allowJsonPaths("$")`), that also doesn't make practical sense, I wonder whether we should actually disallow that? Basically it means "mask nothing" and we already disallow similar situation with `maskKeys(Set.of())` throwing an exception

Reply:
Makes sence. Disallowed $ in allow mode.